python/plugins/processing/algs/qgis/IdwInterpolation.py

```diff
@@ -63,20 +63,61 @@ def getIcon(self):
         return QIcon(os.path.join(pluginPath, 'images', 'interpolation.png'))
 
     def defineCharacteristics(self):
-        self.name, self.i18n_name = self.trAlgorithm('IDW interpolation (using Z-values)')
+        self.name, self.i18n_name = self.trAlgorithm('IDW interpolation')
         self.group, self.i18n_group = self.trAlgorithm('Interpolation')
 
-        self.TYPES = [self.tr('Points'),
-                      self.tr('Structure lines'),
-                      self.tr('Break lines')
-                      ]
-
-        self.addParameter(ParameterVector(self.INPUT_LAYER,
-                                          self.tr('Vector layer')))
-        self.addParameter(ParameterSelection(self.LAYER_TYPE,
-                                             self.tr('Type'),
-                                             self.TYPES,
-                                             0))
+        class ParameterInterpolationData(Parameter):
+            default_metadata = {
+                'widget_wrapper': 'processing.algs.qgis.ui.InterpolationDataWidget.InterpolationDataWidgetWrapper'
+            }
+
+            def __init__(self, name='', description=''):
+                Parameter.__init__(self, name, description)
+
+            def setValue(self, value):
+                if value is None:
+                    if not self.optional:
+                        return False
+                    self.value = None
+                    return True
+
+                if value == '':
+                   if not self.optional:
+                       return False
+
+                if isinstance(value, str):
+                    self.value = value if value != '' else None
+                else:
+                    self.value = ParameterInterpolationData.dataToString(value)
```

Comment:
Would it possibly be nicer to internally keep data and only convert to string when required?